### PR TITLE
compile with nccl2

### LINKED ISSUE #8195: NCCL compilation and linking version not match
Currently, paddle is compiled with nccl1.3 by pulling its source code from https://github.com/NVIDIA/nccl. During the runtime, it is linked to nccl2. This would result in failure such as `unhandled cuda error`.

The CI doesn't fail because some CI machine only have one GPU so the nccl test is skipped.

### PR DESCRIPTION
**Pull request by tonyyang-svail**
fix #8195

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,7 +22,7 @@ COPY ./paddle/scripts/docker/root/ /root/
 
 RUN apt-get update && \
     apt-get install -y \
-    git python-pip python-dev openssh-server bison libnccl-dev \
```
Comment (typhoonzero) [inline Dockerfile]: Should also update `tools/manylinux1/Dockerfile.x64` for release builds.

File: paddle/platform/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 if(WITH_GPU)
-  cc_library(enforce SRCS enforce.cc DEPS nccl)
+  cc_library(enforce SRCS enforce.cc DEPS)
```
Comment (dzhwinter) [inline paddle/platform/CMakeLists.txt]: I vaguely remember that NVIDIA guys have upgraded these things with CUDA9, cudnn7, NCCL2. 
You teach me that we need to call `group_start` before call `ALLReduce` NCCL calls. Is it right?

File: Dockerfile (hunk 1 above)
Comment (tonyyang-svail) [reply Dockerfile]: `libnccl` is not at the `nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04`. So I added it back. In this case, we don't need to update `tools/manylinux1/Dockerfile.x64`. :)

File: paddle/platform/CMakeLists.txt (hunk 2 above)
Comment (tonyyang-svail) [reply paddle/platform/CMakeLists.txt]: Thank you for reminding me. I've added `group_start` and `group_end` in this PR.

**Pull request closed without merging**